src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs

```diff
@@ -212,32 +212,43 @@ private static DirectoryPath GetVisualStudioPath(ICakeEnvironment environment, M
                 "Enterprise",
                 "Professional",
                 "Community",
-                "BuildTools"
+                "BuildTools",
+                "Preview",
```

Comment:
Why add back preview? RC removed the preview folder.

Reply:
The latest Build Tools available for download (`vs_buildtools__1224110841.1636304512.exe`) still use the Preview folder. I installed on my machine first to confirm.

![image](https://user-images.githubusercontent.com/177608/140657027-e5958168-57b5-4b67-8b53-3d8a46d09960.png)

It's also on the screenshot of [discussion 3476](https://github.com/cake-build/cake/discussions/3476#discussioncomment-1595161).

Reply:
I downloaded https://aka.ms/vs/17/release/vs_BuildTools.exe

from https://visualstudio.microsoft.com/downloads/#tools-for-visual-studio-2022-rc-family

<img width="334" alt="image" src="https://user-images.githubusercontent.com/1647294/140658159-d952ee03-5558-41ee-bd98-60516b431a2d.png">

<img width="601" alt="image" src="https://user-images.githubusercontent.com/1647294/140658260-f9e8c960-c5d5-4a99-b034-a1716fe95a74.png">

Reply:
Odd. I downloaded from `https://visualstudio.microsoft.com` as well, but got a different download link - history shows `https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&ch=pre&rel=17`

Do you know if Preview is going to stop working soon and force devs to update to RC or RTM after next week? I'm thinking it doesn't hurt to leave the `Preview` as the last fallback

Reply:
We've not had previews in released studio versions before with default resolvers before.